### DISCUSSION TITLE
Have you implement a c++ version?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [marcomusy/vedo#523] [Ideas]
`vedo` is a very good project. Would you plan to implement a c++ version?

**Comment by marcomusy** [comment]
thanks, It would be nice... but I'm afraid that at present we don't have enough energies to do that... If you volunteer I would definitely support that :)

**Comment by zhang-qiang-github** [comment]
I would like to develop a c++ version. Do you have any suggestion to start this project? Or I just need to translate your code from python to c++?

**Comment by RubendeBruin** [comment]
Hi @zhang-qiang-github , thanks for your offer.

On a suggestion of how to start you first of all need to know **why** you would want to do it. It is quite an endeavor to translate and maintain.

What would be the motivation for a c++ port of vedo? I personally work with both python and c++ where I use c++ for the more performance oriented code and python for everything else.

I think vedo goes very well with the "quick and easy" style of python.  If I want to have something done quickly I will go for python. So for getting things done I would not need a c++ version of vedo.

I would use c++ if I need robustness and speed. In that case I would prefer to interface with vtk directly (more optimized).

I guess the question is: why can you not use the python version of vedo?

If after this demotivational message you still want to give it a go then the next step is to list the features of vedo that you want to port and check if they can be ported. Some of the magic in python (being an interpreted language) is not easy to convert to c++. You probably would want to focus on a somewhat simplified port first.

**Comment by marcomusy** [comment]
I think @RubendeBruin has a very good point here... it's easier said than done.. it would require quite some effort.
Maybe the target would be some user who _doesn't_ want to learn vtk and yet use the vtk functionality in the simpler vedo-like style?
About speed, i guess that depends upon what **else** in your code you do, because calling a vtk filter from the c++ or from the python wrapping of it doesn't change much.
